internal/overlord/servstate/handlers.go

```diff
@@ -187,65 +194,80 @@ func (m *ServiceManager) doStop(task *state.Task, tomb *tomb.Tomb) error {
 		return fmt.Errorf("cannot get service request: %w", err)
 	}
 
-	m.servicesLock.Lock()
-	s := m.services[request.Name]
-	if s == nil || s.getStateLocked() == stateStopped {
-		m.servicesLock.Unlock()
-		m.state.Lock()
-		task.Logf("Service %q already stopped.", request.Name)
-		m.state.Unlock()
+	service := m.serviceForStop(task, request.Name)
+	if service == nil {
 		return nil
 	}
-	m.servicesLock.Unlock()
 
 	// Stop service: send SIGTERM, and if that doesn't stop the process in a
 	// short time, send SIGKILL.
-	err = s.stop()
+	err = service.stop()
 	if err != nil {
 		return err
 	}
 
 	timeout := time.After(failWait + 100*time.Millisecond)
 	for {
 		select {
-		case err := <-s.stopped:
+		case err := <-service.stopped:
 			if err != nil {
 				return fmt.Errorf("cannot stop service: %w", err)
 			}
 			// Stopped successfully.
 			return nil
 		case <-tomb.Dying():
-			// Stop cancelled (shouldn't really happen, so disallow for simplicity).
-			logger.Noticef("Cannot cancel stop (service %q)", request.Name)
+			// Stop operation cancelled (ignore for simplicity).
+			logger.Noticef("Ignoring cancellation of stop for service %q", request.Name)
```

Comment:
It's hard to tell whether you realize or not, but aborting (cancelling) the `start` or `stop` only aborts the starting and stopping "changes" -- it's not referring to aborting the process itself. So here's what I've done:

In `doStop`, for aborting a stop operation, I'm not sure it makes sense to say "Waiting for service %q to stop...", because you've just tried to abort/cancel it from stopping. Aborting a stop would be trying to resuscitate / restart the service, but who knows what state it's in. As your comment mentions, we've already sent the terminate/kill signal at this point, so we can't do much except log it. In any case, I've changed it to the following:

```go
	case <-tomb.Dying():
			// User tried to abort the stop, but SIGTERM and/or SIGKILL have
			// already been sent to the process, so there's not much more we
			// can do than log it.
			logger.Noticef("Cannot abort stop for service %q, kill signals already sent", request.Name)
```

In `doStart`, for aborting a start operation, we can actually abort/cancel the start by killing the process. So I've updated this to send SIGKILL and report the `start` change as having failed. Alternatively, we could just disallow aborting a stop operation (you've only got 1 second to do so!) and log "Cannot abort start for service %q, starting up" or similar.